### PR TITLE
TouchMenu: keep file/exit button on the right

### PR DESCRIPTION
**Pull request by Frenzie**
References #2555.

Before:
screenshot_2017-02-24_16-29-57

After:
screenshot_2017-02-24_16-30-27
screenshot_2017-02-24_16-30-47

### CODE REVIEW

**Comment by Frenzie** [comment]
The after screenshots are a couple of pixels off because I realized I forgot `Screen:scaleBySize(2)`

Screenshot's based on
```lua
self.width - content_width - (math.min(spacing_width, Screen:scaleBySize(20)) * #self.icons *2) 
```

I changed it to
```lua
self.width - content_width - (math.min(spacing_width, Screen:scaleBySize(20)) * #self.icons *2) + Screen:scaleBySize(2),
```

File: frontend/ui/widget/touchmenu.lua [hunk 1]
```diff
@@ -219,6 +219,12 @@ function TouchMenuBar:init()
             for i, sep in ipairs(self.icon_seps) do
                 local current_icon = i == k - 1 or i == k
                 self.icon_seps[i].style = current_icon and "solid" or "none"
+                if i == #self.icon_seps - 1 then
```
Comment (Frenzie) [inline frontend/ui/widget/touchmenu.lua]: If you like this change then I should add a comment explaining what this does first.

Comment (retrue) [inline frontend/ui/widget/touchmenu.lua]: Lol, the screen of the Kobo Mini is so small that the home/exit icon is always on the right, so I won't notice any different.

**Comment by Hzj-jie** [comment]
Looks good. But I do agree the "home" icon is confusion. Several of my friends tried KOReader on my Kobo, the first button they have clicked was the "home". It provides a wrong information as the front page of the app.

**Comment by houqp** [comment]
Do we still need this one after https://github.com/koreader/koreader/pull/2577 is merged?

**Comment by Frenzie** [comment]
Yes, we do. :)

**Comment by houqp** [comment]
Based on the discussion in #2555, @baskerville suggested moving the home icon to the right because it's a uncancellable action. Now that #2577 is merged, this is not the case anymore. So why do we need to move it all the way to the right again? ;P

**Comment by Frenzie** [comment]
Lol

**Comment by Hzj-jie** [comment]
Now,
In file manager, we have history / open last document / version / help and exit in hamburg.
In reader, we have book status / history / version / help / exit in hamburg.

In my opinion, history and open last document look like "search" function, and can be placed in search menu. Book status looks like document feature, can be placed in document menu.
Then we only have version / help / exit, and maybe power off later in hamburg, which seems reasonable to be put on the right.

Well, this is my personal opinion only.

**Comment by Frenzie** [comment]
I'd put open last document in reader as well for consistency.

But search? Is there any precedent for that in other applications?

**Comment by Frenzie** [comment]
Purely UX-wise, I think it's easier both to find and to tap corner items than something in the middle. I imagine that's also just about everything from the Window start button to GNOME to Android puts hamburger dead left or dead right, not somewhere on the middle to the right end of a bunch of other icons. The sole exception is Opera, where I had to go hunting for the freaking hamburger menu. In their defense I suppose fullscreen is frequently on the bottom right. 

Anyway, that particular issue doesn't apply to KOReader as long as we keep it right-most, but I can say from working with this change for the past few days that I really regret not having done this a couple of years ago. I wanted to, but I didn't feel comfortable enough at the time, plus I had different priorities (proper translation). I suggest giving it a go for a few days if you're unsure. ;-)

**Comment by pazos** [comment]
> In my opinion, history and open last document look like "search" function, and can be placed in search menu. Book status looks like document feature, can be placed in document menu.

@Hzj-jie : I agree.

It doesn't need to be the "search" tab, but a more generic file "explorer" tag. AFAIK both options are avaliable only from the file manager app, but not from the reader app.

The reader app UX seems a good start point here. It has two tabs:the bookmark tab and the --reader.app-- tab, which shows config only avaliable from that app at the left of the screen (those are more specific than the settings or the plugins tab). In file manager app the first icon could be the *more related to filemanager as posible* IMO

**Pull request merged**